### ISSUE TITLE
Replace MatcherAssert.assertThat with 'new Assertion<>'

### ISSUE THREAD

**Issue by llorllale** [llorllale/cactoos-matchers#18]
Let's add a substitute for `MatcherAssert.assertThat`, something like `Assertion<T>`:

```java
new Assertion<>(
    "Collection cannot be empty",
    collection.size(),
    new IsNot<>(0)
).verify();    //might throw AssertionError
```

Note: ctor must have the reason text parameter

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/llorllale/cactoos-matchers/issues/18#issuecomment-393205177))

@llorllale Job [#18](https://github.com/llorllale/cactoos-matchers/issues/18) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by llorllale** [comment]
@llorllale let's make `Assertion` accept a `Scalar` of the result to be tested. This way, we can also use it to assert on exceptions. 

It would thus look like this for normal execution flow:
```java
new Assertion<>(
    "Collection cannot be empty",
    () -> collection.size(),
    new IsNot<>(0)
).verify();    //might throw AssertionError
```

To test for exceptions, one would perhaps just provide a matcher for the exception:
```java
new Assertion<>(
    "Collection cannot be empty",
    () -> collection.size(),
    new Throws(IllegalArgumentException.class, "Illegal size")  // exception matcher
).verify();    //might throw AssertionError
```

See #https://github.com/llorllale/cactoos-matchers/issues/40#issuecomment-445421752

**Comment by llorllale** [comment]
@0crat assign me

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/llorllale/cactoos-matchers/issues/18#issuecomment-445478303))

@llorllale @llorllale[/z](https://www.0crat.com/u/llorllale) not enough funds available in the project, can't set budget of job [#18](https://github.com/llorllale/cactoos-matchers/issues/18), see [§21](http://www.zerocracy.com/policy.html#21); @llorllale[/z](https://www.0crat.com/u/llorllale) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
It is strongly discouraged to assign jobs to their creators, see [§19](http://www.zerocracy.com/policy.html#19): -15 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0pdd** [comment]
@llorllale 2 puzzles [#52](https://github.com/llorllale/cactoos-matchers/issues/52), [#53](https://github.com/llorllale/cactoos-matchers/issues/53) are still not solved.

**Comment by llorllale** [comment]
@llorllale `Assertion` implemented in PR #49

**Comment by 0crat** [comment]
Job [#18](https://github.com/llorllale/cactoos-matchers/issues/18) is not in the agenda of @llorllale[/z](https://www.0crat.com/u/llorllale), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#18](https://github.com/llorllale/cactoos-matchers/issues/18) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +35 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0pdd** [comment]
@llorllale 3 puzzles [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81) are still not solved; solved: [#52](https://github.com/llorllale/cactoos-matchers/issues/52).

**Comment by 0pdd** [comment]
@llorllale 3 puzzles [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#84](https://github.com/llorllale/cactoos-matchers/issues/84) are still not solved; solved: [#52](https://github.com/llorllale/cactoos-matchers/issues/52), [#81](https://github.com/llorllale/cactoos-matchers/issues/81).

**Comment by 0pdd** [comment]
@llorllale 3 puzzles [#52](https://github.com/llorllale/cactoos-matchers/issues/52), [#88](https://github.com/llorllale/cactoos-matchers/issues/88), [#99](https://github.com/llorllale/cactoos-matchers/issues/99) are still not solved; solved: [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81), [#84](https://github.com/llorllale/cactoos-matchers/issues/84).

**Comment by 0pdd** [comment]
@llorllale 2 puzzles [#52](https://github.com/llorllale/cactoos-matchers/issues/52), [#88](https://github.com/llorllale/cactoos-matchers/issues/88) are still not solved; solved: [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81), [#84](https://github.com/llorllale/cactoos-matchers/issues/84), [#99](https://github.com/llorllale/cactoos-matchers/issues/99).

**Comment by 0pdd** [comment]
@llorllale 2 puzzles [#110](https://github.com/llorllale/cactoos-matchers/issues/110), [#52](https://github.com/llorllale/cactoos-matchers/issues/52) are still not solved; solved: [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81), [#84](https://github.com/llorllale/cactoos-matchers/issues/84), [#88](https://github.com/llorllale/cactoos-matchers/issues/88), [#99](https://github.com/llorllale/cactoos-matchers/issues/99).

**Comment by 0pdd** [comment]
@llorllale 2 puzzles [#113](https://github.com/llorllale/cactoos-matchers/issues/113), [#52](https://github.com/llorllale/cactoos-matchers/issues/52) are still not solved; solved: [#110](https://github.com/llorllale/cactoos-matchers/issues/110), [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81), [#84](https://github.com/llorllale/cactoos-matchers/issues/84), [#88](https://github.com/llorllale/cactoos-matchers/issues/88), [#99](https://github.com/llorllale/cactoos-matchers/issues/99).

**Comment by 0pdd** [comment]
@llorllale 2 puzzles [#116](https://github.com/llorllale/cactoos-matchers/issues/116), [#52](https://github.com/llorllale/cactoos-matchers/issues/52) are still not solved; solved: [#110](https://github.com/llorllale/cactoos-matchers/issues/110), [#113](https://github.com/llorllale/cactoos-matchers/issues/113), [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81), [#84](https://github.com/llorllale/cactoos-matchers/issues/84), [#88](https://github.com/llorllale/cactoos-matchers/issues/88), [#99](https://github.com/llorllale/cactoos-matchers/issues/99).

**Comment by 0pdd** [comment]
@llorllale the puzzle [#52](https://github.com/llorllale/cactoos-matchers/issues/52) is still not solved; solved: [#110](https://github.com/llorllale/cactoos-matchers/issues/110), [#113](https://github.com/llorllale/cactoos-matchers/issues/113), [#116](https://github.com/llorllale/cactoos-matchers/issues/116), [#53](https://github.com/llorllale/cactoos-matchers/issues/53), [#80](https://github.com/llorllale/cactoos-matchers/issues/80), [#81](https://github.com/llorllale/cactoos-matchers/issues/81), [#84](https://github.com/llorllale/cactoos-matchers/issues/84), [#88](https://github.com/llorllale/cactoos-matchers/issues/88), [#99](https://github.com/llorllale/cactoos-matchers/issues/99).
